lib/open_project/text_formatting/filters/markdown_filter.rb

```diff
@@ -31,35 +31,36 @@ module Filters
     class MarkdownFilter < HTML::Pipeline::MarkdownFilter
       # Convert Markdown to HTML using CommonMarker
       def call
-        render_html parse
+        render_html
       end
 
       private
 
       ##
-      # Get initial CommonMarker AST for further processing
-      #
-      def parse
-        parse_options = %i[LIBERAL_HTML_TAG STRIKETHROUGH_DOUBLE_TILDE UNSAFE]
-
-        # We need liberal html tags thus parsing and rendering are several steps
-        # Check: We may be able to reuse the ast instead of rendering to html and then parsing with nokogiri again.
-        CommonMarker.render_doc(
-          text,
-          parse_options,
-          commonmark_extensions
-        )
+      # Render markdown to html
+      def render_html
+        Commonmarker.to_html(text, options: commonmarker_options, plugins: commonmarker_plugins)
+                    .tap(&:rstrip!)
       end
 
       ##
-      # Render the transformed AST
-      def render_html(ast)
-        render_options = %i[GITHUB_PRE_LANG UNSAFE]
-        render_options << :HARDBREAKS if context[:gfm] != false
+      # CommonMarker Options
+      # https://github.com/gjtorikian/commonmarker#options
+      def commonmarker_options
+        {
+          parse: { smart: false },
+          extension: commonmark_extensions.map { |k| [k, true] }.to_h,
```

Comment:
The simpler way of doing this is `commonmark_extensions.index_with(true)`

Reply:
I'm amazed how many elegant helper ruby/rails has for data wrangling. Thanks for introducing me to this one!